### PR TITLE
Feature/2668 new interop submission

### PR DESCRIPTION
**Pull request by nickguendling**
## Description
This implements https://jira.itc.sap.com/browse/EXPOSUREAPP-2668 and shows the participating countries right before the submission.

This also contains the first steps to the promised refactoring of the exposure submission coordinator and adds a model to it.

IMG_DC04B7DD74FB-1
IMG_807142F0713A-1

### CODE REVIEW

**Review by janetback — APPROVED** [review]
UA approved

**Review by SwissMountainFrog — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionWarnOthersViewController.swift [hunk 1]
```diff
@@ -73,62 +76,7 @@ class ExposureSubmissionWarnOthersViewController: DynamicTableViewController, EN
 			forCellReuseIdentifier: CustomCellReuseIdentifiers.roundedCell.rawValue
 		)
 
-		dynamicTableViewModel = dynamicTableViewModel()
-	}
-
-	private func dynamicTableViewModel() -> DynamicTableViewModel {
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionWarnOthersViewController.swift:79]: ❤️

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/__tests__/ExposureSubmissionUITests.swift [hunk 2]
```diff
@@ -277,108 +267,10 @@ class ENAUITests_04_ExposureSubmissionUITests: XCTestCase {
 		XCTAssertTrue(app.navigationBars["ENA.ExposureSubmissionWarnOthersView"].waitForExistence(timeout: .medium))
 		app.buttons["AppStrings.ExposureSubmission.continueText"].tap()
 
-	#if INTEROP
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/__tests__/ExposureSubmissionUITests.swift:280]: Hmm shouldn't we remove the `INTEROP` from the Test targets? And shouldn't it everything in between ```
#if INTEROP
#endif
``` be the part we keep?

Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/__tests__/ExposureSubmissionUITests.swift:280]: These were UI tests that tested the old interoperability screens that have been scrubbed, so I had to remove the screens as well as the tests.

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionWarnOthersViewController.swift (hunk 1 above)
Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionWarnOthersViewController.swift:79]: 😘

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinatorModel.swift [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+		appConfigurationProvider.appConfiguration { applicationConfiguration in
+			isLoading(false)
+
+			guard let supportedCountries = applicationConfiguration?.supportedCountries.compactMap({ Country(countryCode: $0) }) else {
+				onError(.noAppConfiguration)
+				return
+			}
+
+			if supportedCountries.isEmpty {
+				self.supportedCountries = ["DE"].compactMap { Country(countryCode: $0) }
```
Comment (johannesrohwer) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinatorModel.swift:122]: would DE also be available via Country.defaultCountry() ?

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/__tests__/ExposureSubmissionUITests.swift (hunk 2 above)
Comment (marcussc) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/__tests__/ExposureSubmissionUITests.swift:280]: Ah okay. Thanks!

**Review by marcussc — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationViewController.swift [hunk 4]
```diff
@@ -34,7 +34,7 @@ class AppInformationViewController: DynamicTableViewController {
 			.section(
 				header: .space(height: 32),
 				footer: .view(footerView()),
-				separators: false,
+				separators: .none,
```
Comment (johannesrohwer) [inline src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationViewController.swift:37]: very nice to replace the bool with an adequate case here! 🏅

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionWarnOthersViewModel.swift [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+import UIKit
+
+struct ExposureSubmissionWarnOthersViewModel {
+	
+	// MARK: - Init
+
+	init(
+		supportedCountries: [Country]
+	) {
+		self.supportedCountries = supportedCountries.sorted { $0.localizedName.localizedCompare($1.localizedName) == .orderedAscending }
```
Comment (johannesrohwer) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionWarnOthersViewModel.swift:28]: nice to see localized comparison here :)

**Review by johannesrohwer — APPROVED** [review]
great job! 🎉  I like the new ExposureSubmissionCoordinatorModel, it really helps to keep things clean!

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinatorModel.swift (hunk 3 above)
Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinatorModel.swift]: Thanks, that's right, totally makes sense. Replaced it 👍

File: src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationViewController.swift (hunk 4 above)
Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Scenes/AppInformation/AppInformationViewController.swift:37]: 😁

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionWarnOthersViewModel.swift (hunk 5 above)
Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/ExposureSubmissionWarnOthersViewModel.swift:28]: Luckily I initially tested it with Dänemark and Deutschland and was surprised that Deutschland appeared first using non-localized sorting 😅

**Pull request merged**